clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1064,6 +1058,17 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
     }
   }
 
+  if (getLangOpts().SYCLIsDevice) {
+    if (const FunctionDecl *FD = dyn_cast_or_null<FunctionDecl>(D)) {
```

Comment:
Is this line necessary?  Decl has a hasAttr function as well, so I don't think we care that this is a function?  we perahps care that "D" is not null, but we don't care that it is a FunctionDecl.